src/library/searchquery.cpp

```diff
@@ -20,7 +22,9 @@ QVariant getTrackValueForColumn(const TrackPointer& pTrack, const QString& colum
     } else if (column == LIBRARYTABLE_ALBUMARTIST) {
         return pTrack->getAlbumArtist();
     } else if (column == LIBRARYTABLE_YEAR) {
-        return pTrack->getYear();
+        // We use only the year that is part of the first four digits
+        // In all possible formats.
+        return pTrack->getYear().left(4);
```

Comment:
shouldn't we somehow fix this on the metadata parsing level? The primary issue is IMO that getting the year from the track doesn't get us its year but possibly its exact date. Saving the date as the year and the working around that in the DB query code does not seem like the right approach to me.

Reply:
The underlying issue is the history of ID3. Original this was only a year field, but some have started to us it for full dates until other fields have been created. In Mixxx we have only the year column and parse and copy the data the best we can, see:  
https://github.com/mixxxdj/mixxx/blob/2fe0fc58b1be4813b6a3f73288d3468eb775ba6b/src/track/taglib/trackmetadata_id3v2.cpp#L798-L824
So it is finally a badly named date column, but it can contain any string unfortunately. 

The good think is that in all usable cases the four first characters contain the year. This PR makes use of it.   

Do you have a different fix in mind that suits for a stable release?

Reply:
I see, well then the issue boils down to naming IMO. "just refactor" all occurrences of `year` with date and then keep the special query parser that derives the year from the date.

Reply:
I am not in favor to do it because the library column is also called "year". This is IMHO OK,, because in my library the vast majority of tracks have only a year. Using the column as date is the exception. 

Currently we use consistently the term year changing this would be a big refactoring. I am not total against doing it, but that should be done on a broad consens and not in a bug-fix PR.

Reply:
> because in my library the vast majority of tracks have only a year. Using the column as date is the exception.

I'm not quite sure but I think some DAWs and possibly bandcamp started adding a `YY-MM-DD` by default, so I question the rarity of that. 

> Currently we use consistently the term year changing this would be a big refactoring. I am not total against doing it, but that should be done on a broad consens and not in a bug-fix PR.

Sure, then at least document the misnomer so people don't get confused when `Track::getYear` gives them a string which can't be cleanly parsed into an actual year number.

Reply:
> This is IMHO OK,, because in my library the vast majority of tracks have only a year. Using the column as date is the exception.

This might be an exception in your personal library. But even MusicBrainz Picard stores dates.